### PR TITLE
Scope time domain measurement

### PR DESCRIPTION
**Pull request by Jay-Cao123**
### What does this Pull Request accomplish?

- This measurement includes a LabVIEW project and measurement service library that automatically generated by 'Measurement Plug-in'.
- The VIs in the library are updated to support NI-SCOPE Time Domain measurements.
- A TestStand sequence file is included to show how to configure the pin map, register instrument sessions with the session management service, and run a measurement.
- InstrumentStudio files are included to show the manual testing.
- A Pin map file is included.
- A readme file is included.

### Why should this Pull Request be merged?

Adding this as one of the PCBA Function Test measurements.

### What testing has been done?

Test setup:

NI PXIe-5122 (SCOPE1) CH1 is connected to NI PXIe-4071 (DAQmx1) AO1.
The output signal of AO1: Voltage sinewave, 2 volts amplitude, 1Hz.
image

Test result:
InstrumentStudio:
image
TestStand:
image
Build result:
image

### CODE REVIEW

**Comment by Jay-Cao123** [comment]
If the inputs of "Minimum Record Length" is too small that can't cover a whole period waveform, then it will raise this error and can't calculate the frequency or period:

> niScope Fetch Measurement Stats.vi:4490003<ERR>Unable to perform desired measurement operation.  Waveform did not cross the middle reference level at least 3 times. Check the low, mid, and high reference levels. If you are reading a waveform that has a small amplitude, you can use the attribute Hysteresis Percent to reduce the hysteresis thresholds.

image

File: src/NI-SCOPE Time Domain/NIScopeTimeDomain/Measurement Logic.vi
Comment (nathan-murphy) [inline src/NI-SCOPE Time Domain/NIScopeTimeDomain/Measurement Logic.vi:1]: Please remove any of the #ToDo comments in this block diagram, and others. You can use the Bookmark Manager to find all of them. Please make sure you're actually 'do'ing the things in the instructions, too.

Comment (nathan-murphy) [inline src/NI-SCOPE Time Domain/NIScopeTimeDomain/Measurement Logic.vi:1]: image
Remove the breakpoint on this VI (and anywhere else) before committing code.

File: src/NI-SCOPE Time Domain/NIScopeTimeDomain/Measurement UI.vi
Comment (nathan-murphy) [inline src/NI-SCOPE Time Domain/NIScopeTimeDomain/Measurement UI.vi:1]: Change minimum sample rate to use Engineering Notation instead of Scientific.

Comment (nathan-murphy) [reply src/NI-SCOPE Time Domain/NIScopeTimeDomain/Measurement UI.vi:1]: UI should be aligned across ALL measurements. For example, Font Size, title names, etc. Please work together to come up with common phrasing, font sizes, etc.

**Review by nathan-murphy — CHANGES_REQUESTED** [review]

File: src/NI-SCOPE Time Domain/NIScopeTimeDomain/Get Type Specializations.vi
Comment (kena0405) [inline src/NI-SCOPE Time Domain/NIScopeTimeDomain/Get Type Specializations.vi:1]: image
'niScope' is missing.

File: src/NI-SCOPE Time Domain/NIScopeTimeDomain/Measurement UI.vi
Comment (kena0405) [reply src/NI-SCOPE Time Domain/NIScopeTimeDomain/Measurement UI.vi:1]: image

Right-click and untick the 'View As Icon ' to align with the right column.

File: test/TestStand/NIScopeTimeDomain/NIScopeTimeDomain.seq
Comment (kena0405) [inline test/TestStand/NIScopeTimeDomain/NIScopeTimeDomain.seq:1]: Reselect the project
image

**Review by kirstyong — CHANGES_REQUESTED** [review]
@Jay-Cao123 did you validate that the exe builds fine? Could you add that to your testing notes in the PR description if you did so.

**Review by kena0405 — APPROVED** [review]

File: src/NI-SCOPE Time Domain/NIScopeTimeDomain/Measurement UI.vi
Comment (nathan-murphy) [inline src/NI-SCOPE Time Domain/NIScopeTimeDomain/Measurement UI.vi:1]: Graph title needs to be **bold** based on the design spec

Comment (nathan-murphy) [inline src/NI-SCOPE Time Domain/NIScopeTimeDomain/Measurement UI.vi:1]: image
Left-align the Trigger Coupling label

File: src/NI-SCOPE Time Domain/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+This example performs time domain measurement using NI SCOPE.
+
+
+### Features
+
+- Uses the NI-SCOPE LabVIEW API.
+- Pin-aware, supporting one session and one pin.
+  - Uses the same selected measurement function and range for all selected pin/site combinations.
+- Includes InstrumentStudio project files.
+- Includes a TestStand sequence showing how to configure the pin map, register.
```
Comment (nathan-murphy) [inline src/NI-SCOPE Time Domain/README.md]: remove the period at the end of this line, and then lines 12 and 13 should probably be on one line.

File: src/NI-SCOPE Time Domain/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  instrument sessions with the session management service, and run a measurement.
+  - For the sake of simplicity, the TestStand sequence handles pin map and session registration and unregistration in the `Setup` and `Cleanup` sections of the main sequence. For **Test UUTs** and batch process model use cases, these steps should be moved to the `ProcessSetup` and `ProcessCleanup` callbacks.
```
Comment (nathan-murphy) [inline src/NI-SCOPE Time Domain/README.md]: Test UUTs is not a process model. This should probably say: For **Sequential** and **Batch** process models, ...

File: src/NI-SCOPE Time Domain/NIScopeTimeDomain.lvproj [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+				<Property Name="Bld_excludeLibraryItems" Type="Bool">true</Property>
+				<Property Name="Bld_excludePolymorphicVIs" Type="Bool">true</Property>
+				<Property Name="Bld_localDestDir" Type="Path">../builds/NI_AB_PROJECTNAME/NIScopeTimeDomain</Property>
+				<Property Name="Bld_localDestDirType" Type="Str">relativeToCommon</Property>
+				<Property Name="Bld_modifyLibraryFile" Type="Bool">true</Property>
+				<Property Name="Bld_postActionVIID" Type="Ref">/My Computer/NIScopeTimeDomain.lvlib/Advanced/Build Assets/Post-Build Action.vi</Property>
+				<Property Name="Bld_previewCacheID" Type="Str">{4F808489-7F5C-4F6B-BB9E-6C94ED82B21A}</Property>
+				<Property Name="Bld_version.major" Type="Int">1</Property>
+				<Property Name="Destination[0].destName" Type="Str">NIScopeTimeDomain.exe</Property>
+				<Property Name="Destination[0].path" Type="Path">../builds/NI_AB_PROJECTNAME/NIScopeTimeDomain/NIScopeTimeDomain.exe</Property>
```
Comment (nathan-murphy) [inline src/NI-SCOPE Time Domain/NIScopeTimeDomain.lvproj]: `builds` directory should either: 
- not be placed under `src` directory
- be added to the `.gitignore` file as an ignored directory

@kirstyong what do you think?

File: src/NI-SCOPE Time Domain/NIScopeTimeDomain/Measurement Configuration.ctl
Comment (nathan-murphy) [inline src/NI-SCOPE Time Domain/NIScopeTimeDomain/Measurement Configuration.ctl:1]: Can you please make sure the default values in this `ctl` match the values in the UI? It's not clear which value you intended to make default if they are not the same:
image

**Review by nathan-murphy — CHANGES_REQUESTED** [review]

File: src/NI-SCOPE Time Domain/NIScopeTimeDomain.lvproj (hunk 3 above)
Comment (kirstyong) [reply src/NI-SCOPE Time Domain/NIScopeTimeDomain.lvproj]: I'd prefer they be added to .gitignore. That builds path under src is just the default path that LabVIEW comes with. I'm ok with how it is as is and I'll update the gitignore in a separate PR.

**Review by nathan-murphy — APPROVED** [review]
Looks good!

File: src/NIScopeTimeDomain/NIScopeTimeDomain/Measurement UI.vi
Comment (kirstyong) [inline src/NIScopeTimeDomain/NIScopeTimeDomain/Measurement UI.vi:1]: My final ask.....please update this header to say "Pin Settings" so we can be consistent with the other measurements. :)
image

**Review by kirstyong — APPROVED** [review]
Going to set this to approve. Just need the "Pin Settings" header updated is all. Great work!

**Pull request merged**